test/integration/controllermanager/controllerregistration/controllerregistration_test.go

```diff
@@ -219,6 +240,7 @@ var _ = Describe("ControllerRegistration controller test", func() {
 
 			By("Create ControllerRegistration")
 			Expect(testClient.Create(ctx, controllerRegistration)).To(Succeed())
```

Comment:
Shouldn't there be a `DeferCleanup()` call after this creation, so that we delete the `ControllerRegistration` in case the tests errors before [line 278](https://github.com/gardener/gardener/blob/5f1c4b89366c92f7fec8999a6fb6b415acdbc327/test/integration/controllermanager/controllerregistration/controllerregistration_test.go#L278)

I think there are a few similar cases in this test in other places.